rs-commerce/src/views/user-profile/change-name/ChangeName.tsx

```diff
@@ -0,0 +1,172 @@
+      } else {
+        setError(errorMsg);
+        dispatch(updateProfileActions.setCheckNewName(false));
+      }
+    } else {
+      const limitData = new Date();
+      limitData.setFullYear(limitData.getFullYear() - 13);
+      const birthdayData = new Date(value);
+      if (birthdayData <= limitData) {
+        setError('');
+        if (!firstNameError && !lastNameError && !emailError && !dateOfBirthError)
+          dispatch(updateProfileActions.setCheckNewName(true));
+      } else {
+        setError(errorMsg);
+        dispatch(updateProfileActions.setCheckNewName(false));
+      }
+    }
+  };
+
+  const inputArrayAddress: InputProfileType[] = [
+    {
+      title: 'First name',
+      id: 'first-name',
+      isSizeSmall: false,
+      type: 'text',
+      isDisabled: false,
+      errorMsg: firstNameError,
+      value: firstName,
+      handler: (e) =>
+        checkField(e, setFirstName, setFirstNameError, regExpObj.firstName, errorMsgObj.firstName),
+    },
+    {
+      title: 'Last name',
+      id: 'last-name',
+      isSizeSmall: false,
+      type: 'text',
+      isDisabled: false,
+      errorMsg: lastNameError,
+      value: lastName,
+      handler: (e) =>
+        checkField(e, setLastName, setLastNameError, regExpObj.lastName, errorMsgObj.lastName),
+    },
+    {
+      title: 'Date of birth',
+      id: 'date-of-birth',
+      isSizeSmall: false,
+      type: 'date',
+      isDisabled: false,
+      value: dateOfBirth,
+      errorMsg: dateOfBirthError,
+      handler: (e) =>
+        checkField(
+          e,
+          setDateOfBirth,
+          setDateOfBirthError,
+          regExpObj.lastName,
+          errorMsgObj.dateOfBirth,
+        ),
+    },
+    {
+      title: 'Email',
+      id: 'email',
+      isSizeSmall: false,
+      type: 'email',
+      isDisabled: false,
+      value: email,
+      errorMsg: emailError,
+      handler: (e) => checkField(e, setEmail, setEmailError, regExpObj.email, errorMsgObj.email),
+    },
+  ];
+
+  const clearFieldsOnPage = (): void => {
+    setFirstName('');
+    setLastName('');
+    setDateOfBirth('');
+    setEmail('');
+    dispatch(updateProfileActions.setCheckNewName(false));
+  };
+
+  const saveBtnClick = (): void => {
+    console.log('save parameters');
+    if (userData) {
+      apiRootWithExistingTokenFlow()
+        .customers()
+        .withId({ ID: userData.customer.id })
+        .post({
+          body: {
+            version: userData.customer.version,
+            actions: [
+              {
+                action: 'setFirstName',
+                firstName,
+              },
+              {
+                action: 'setLastName',
+                lastName,
+              },
+              {
+                action: 'setDateOfBirth',
+                dateOfBirth,
+              },
+              {
+                action: 'changeEmail',
+                email,
+              },
+            ],
+          },
+        })
+        .execute();
+    }
```

Comment:
Add message about successfull changing data and about error. After success update current User data on new data in store